src/Chapter2.hs

```diff
@@ -336,7 +353,9 @@ from it!
 ghci> :l src/Chapter2.hs
 -}
 subList :: Int -> Int -> [a] -> [a]
-subList = error "subList: Not implemented!"
+subList start end xs
+  | start >= 0 && end >= 0 = (take (end - start + 1) . drop start) xs
```

Comment:
Your implementation is correct when the right bound is lower than the left. But I think you can implement it more efficiently by adding one more check for this special case :slightly_smiling_face:

Reply:
Ack, I see because im wastefully dropping in that case - revised to:
```
subList :: Int -> Int -> [a] -> [a]
subList start end xs
  | end < start = []
  | start >= 0 && end >= 0 = (take (end - start + 1) . drop start) xs
  | otherwise = []
```